extensions/sql-migration/src/wizard/databaseBackupPage.ts

```diff
@@ -1608,7 +1624,30 @@ export class DatabaseBackupPage extends MigrationWizardPage {
 			this._databaseTable.onCellAction!(
 				async (rowState: azdata.ICellActionEventArgs) => {
 					const buttonState = <azdata.ICellActionEventArgs>rowState;
+					// Launch schema migration for database
 					if (buttonState?.column === 2) {
```

Comment:
so the migrate schema button will appear for all databases, even if the schema has already been migrated? Is there any easy client-side check to disable this if the target already has a schema?

Reply:
It is hard to detect whether schema objects have been migrated. The generate script for each object starts with "If not exist...."

Reply:
fair, but isn't it possible to get the table schema via T-SQL query? Even if we don't know where it came from, if we see that a target schema already exists, then we should block an attempt to do a schema migration right?

Reply:
actually even if that's possible, in this case I don't know how easy it is to query the target directly (at least querying the source is straightforward), so maybe this isn't that feasible

Reply:
We have no way of knowing if their schema is migrated. We can check if they have tables but schema is a lot more objects than that. Would need to run a full object comparison.

Reply:
Ah I didn't see the replies before. Yeah the nuget he is building handles already existing objects. You can proveed with migration as long as there's no schema migration in progress

Reply:
querying target should be easy. think we already do that for collation checks. 

do we have any schema checks in the validator does? if so, can we bring any of them up to the client side

also we should be able to tell if the schema has not been migrated at all (i.e. no tables on the target), in that case we should def warn the cx?

Reply:
We are checking in the table selection dialog, but it's true they could just click migrate on the schema without thinking causing them to have to wait